src/test/scala/org/ergoplatform/nodeView/state/ErgoStateSpecification.scala

```diff
@@ -141,4 +143,36 @@ class ErgoStateSpecification extends ErgoPropertyTest {
     }
   }
 
+  property("ErgoState.execTransactions()") {
+    val bh = BoxHolder(genesisBoxes)
+    val txs = (1 to 15).foldLeft(Vector.empty[ErgoTransaction]) { case (txAcc, _) =>
+      val (transactions, _) = validTransactionsFromBoxes(10000, bh.boxes.values.toSeq, new Random)
+      val allBoxIds = bh.boxes.keys.toSet
+      val txsFromBoxesOnly = transactions.filter { tx =>
+        tx.inputs.map(i => ByteArrayWrapper(i.boxId)).forall(allBoxIds.contains) &&
+          tx.dataInputs.map(i => ByteArrayWrapper(i.boxId)).forall(allBoxIds.contains)
+      }
+      txAcc ++ txsFromBoxesOnly
+    }
+
+    val boxes = bh.boxes
+    val stateContext = emptyStateContext
+
+    // successful validation
+    assert(ErgoState.execTransactions(txs, stateContext)(id => Try(boxes(ByteArrayWrapper(id)))).isValid)
+
+    // missing box in state
+    ErgoState.execTransactions(txs, stateContext)(_ => Failure(new RuntimeException)).errors.head.message shouldBe
+      "Every input of the transaction should be in UTXO. null"
+
+    // tx validation should kick in and detect block height violation
+    val invalidTx = invalidErgoTransactionGen.sample.get
+    assert(
+      ErgoState.execTransactions(txs :+ invalidTx, stateContext)(id => Try(boxes.getOrElse(ByteArrayWrapper(id), invalidTx.outputs.head)))
+        .errors.head.message.startsWith("Transaction outputs should have creationHeight not exceeding block height.")
+    )
+
+    // no transactions are valid
+    assert(ErgoState.execTransactions(Seq.empty, stateContext)(id => Try(boxes(ByteArrayWrapper(id)))).isValid)
```

Comment:
Please add tests and test vectors for cost computation.

Reply:
Ok, I wanted to but the costs are all randomly generated in the helper methods, I have to figure out how to do it so it is not mess